File: store/src/main/java/com/min/store/product/dto/response/ProductResponseDto.java
package com.min.store.product.dto.response;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.min.store.product.domain.ProductId;
import lombok.*;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class ProductResponseDto {
    private ProductId productId;
    private String name;
    private Long price;
    private Long orderPrice;
    private int quantity;
}

File: store/store-product-api/src/main/kotlin/com/store/product/api/dto/ProductResponseDto.kt
package com.store.product.api.dto

import com.store.product.api.domain.ProductDomain

class ProductResponseDto (
    val id: Long?,
    val name: String?,
    val price: Long?
    ) {

    companion object {
        fun from(productDomain: ProductDomain): ProductResponseDto {
            return ProductResponseDto(id = productDomain.id, name = productDomain.name, price = productDomain.price)
        }
    }
}
